### DISCUSSION TITLE
How to prepare Custom Sentence Segmentation fuction?

### DISCUSSION THREAD

**Discussion by karndeepsingh** [explosion/spaCy#10289] [Help: Coding & Implementations]
Hi,
I want to split the paragraphs into sentence and I am using default "sentencizer" in my pipeline and it splits into sentence whereever it finds full stop. But in my text full stop is present inside the word and it splits from there as well. For example:

I have this sentence:
```
DEFINITIONS Words used in nvultiple sections of this document are defined below and other words are defined in Sections 3, 11, 13, 18, 20 and 21. Certain rales regarding the usage of words used in this document are also provided in Sestion 16, (A) "Security Lastrament™ means this document, which is dated WULY 31, 2021 y together with a Riders to this document. (8) "Borrower" is Alejandra Silvera and Trevor Andel Silvera, Wife and Husband, As Coramunity Property with Right of Survivorship BORROWER\'S ADDRESS IS 28778 MEADOWGRASS DR, CASTAIC, CALIFORNIA 91384, Borrower is the trustor under this Security Instrument. (C) "Lender is United Wholesale Mortgage, LLC Lender is a LIMITED LIABILITY COMPANY organized and existing under the laws of MICHIGAN . Lender\'s address is 585 South Boulevard E, Pontiac, Michigan 48341 (D) "Trustee" is David E. Fenne, Esq., Northwest Trustee Services 1241 East Dyer Road, Suite 250, Santa Ana, California 92705 

```
After applying default sentence splitter pipeline, it splits as mentioned below:

```
DEFINITIONS Words used in nvultiple sections of this document are defined below and other words are defined in Sections 3, 11, 13, 18, 20 and 21.
Certain rales regarding the usage of words used in this document are also provided in Sestion 16, (A) "Security Lastrament™ means this document, which is dated WULY 31, 2021 y together with a Riders to this document. (
8) "Borrower" is Alejandra Silvera and Trevor Andel Silvera, Wife and Husband, As Coramunity Property with Right of Survivorship BORROWER'S ADDRESS IS 28778 MEADOWGRASS DR, CASTAIC, CALIFORNIA 91384, Borrower is the trustor under this Security Instrument. (
C) "Lender is United Wholesale Mortgage, LLC Lender is a LIMITED LIABILITY COMPANY organized and existing under the laws of MICHIGAN .
Lender's address is 585 South Boulevard E, Pontiac, Michigan 48341 (D) "Trustee" is David E.
Fenne, Esq.,
Northwest Trustee Services 1241 East Dyer Road, Suite 250, Santa Ana, California 92705

```
If you notice, line " (D) "Trustee" is David E.Fenne, Esq.,"  got split into Trustee" is David E. as one sentence and Fenne, Esq., as another and likewise others.
 I want it to split the sentence where it finds "( ) " for example wherever it finds "(A)" or "(B)" or "(C)" it should split the sentence from there not by using full stop. Because "full stop" is also present in the name of the person and it splitting from there. 

Please, help me how I can build a custom sentence splitter function to get the desired output. 

Thanks

**Comment by thomashacker** [comment]
Hello,
the [sentencizer](https://spacy.io/api/sentencizer) component uses a rule-based approach with the goal to generate sentences fast. If you're using a custom tokenizer, this could be the cause of your issue. However, I would recommend to use the [senter](https://spacy.io/api/sentencerecognizer) component for more precision instead.
